staging/src/k8s.io/kubelet/config/v1beta1/types.go

```diff
@@ -482,6 +482,12 @@ type KubeletConfiguration struct {
 	// Default: true
 	// +optional
 	SerializeImagePulls *bool `json:"serializeImagePulls,omitempty"`
+	// MaxParallelImagePulls sets the maximum number of image pulls in parallel.
+	// This field is ignored if SerializeImagePulls is true.
+	// Setting it to 0 means no limit.
+	// Default: 0
```

Comment:
just thinking out loud, if one wants to run kubelet in a mode when all pods has an image pull policy to be `Never`, would they want this to be `0` indicating "no pulls, please"?

Reply:
If `0` means `no pulls, please`, we should set a negative number for no limit.

We have discussed how this feature goes with promoting parallel pull images.
I supposed it should be something like 
- now(alpha): `SerializeImagePulls=true` and the true limit is 0(for unlimit). 
- later(beta): `SerializeImagePulls=false` and the limit should be 5 by default.(registryPullQPS is by default 5)


If `0` means `no pulls, please`, we should re-think about the default value:
I supposed it should be something like 
- now(alpha): `SerializeImagePulls=true` and the true limit is 1 which is the same meaning as `SerializeImagePulls=true`. 
- later(beta): `SerializeImagePulls=false` and the limit should be 5 by default.(registryPullQPS is by default 5)

Reply:
> Setting it to 0 means no limit.

setting it to null should mean no limit. avoid using a max of 0 when we can simply omit the max field value instead

I think this is how we would specify various options:

no parallel pulls:
```yaml
serializeImagePulls: true # optional, this is the default
maxParallelImagePulls: null
```

parallel pulls with no limit:
```yaml
serializeImagePulls: false
maxParallelImagePulls: null
```

max of 2 images to be pulled in parallel at once:
```yaml
serializeImagePulls: false # optional, should be able to be inferred from non-nil maxParallelImagePulls
maxParallelImagePulls: 2
```

I don't think we would change the default of serializeImagePulls in v1beta1 kubelet config file... we could change it in a future v1 version

Reply:
makes sense. I will change the validation to what Jordan mentioned above, plus ensuring maxParallelImagePulls >= 2.